### ISSUE TITLE
Working with images is great, but pdf does not work

### ISSUE THREAD

**Issue by chrischtili** [imanoop7/Ollama-OCR#5]
`PIL.UnidentifiedImageError: cannot identify image file UploadedFile(file_id='d6c294fe-9942-451e-b371-f08af26327b9', name='IMG_677922CC6FA1-1.pdf', type='application/pdf', size=703795, _file_urls=file_id: "d6c294fe-9942-451e-b371-f08af26327b9" upload_url: "/_stcore/upload_file/b05cd0b0-4e16-4fd6-9c1d-2e5d963d47b7/d6c294fe-9942-451e-b371-f08af26327b9" delete_url: "/_stcore/upload_file/b05cd0b0-4e16-4fd6-9c1d-2e5d963d47b7/d6c294fe-9942-451e-b371-f08af26327b9" )
Traceback:
File "/Users/christian/git/Ollama-OCR/src/ollama_ocr/app.py", line 247, in <module>
    main()
File "/Users/christian/git/Ollama-OCR/src/ollama_ocr/app.py", line 163, in main
    image = Image.open(uploaded_file)
File "/Users/christian/Library/Python/3.9/lib/python/site-packages/PIL/Image.py", line 3536, in open
    raise UnidentifiedImageError(msg)`

**Comment by imanoop7** [comment]
I have tested it with different PDF, it is working.
try with any other PDF file, if issue persist- comment

**Comment by chrischtili** [comment]
No matter which PDF I upload, I always get this error

**Comment by imanoop7** [comment]
will check on this

**Comment by tin2tin** [comment]
Same

**Comment by mmol67** [comment]
Same thing using GUI.
`
PIL.UnidentifiedImageError: cannot identify image file UploadedFile(file_id='30fc1dda-78c8-44d8-9d81-5648e65d048b', name='gretel-clean.pdf', type='application/pdf', size=1063028, _file_urls=file_id: "30fc1dda-78c8-44d8-9d81-5648e65d048b" upload_url: "/_stcore/upload_file/070663d2-b4c6-496d-930a-8c58a339e3ed/30fc1dda-78c8-44d8-9d81-5648e65d048b" delete_url: "/_stcore/upload_file/070663d2-b4c6-496d-930a-8c58a339e3ed/30fc1dda-78c8-44d8-9d81-5648e65d048b" )`

```
Traceback:

File "/home/miguel/bin/Ollama-OCR/src/ollama_ocr/app.py", line 260, in <module>
    main()
File "/home/miguel/bin/Ollama-OCR/src/ollama_ocr/app.py", line 175, in main
    image = Image.open(uploaded_file)
File "/home/miguel/ollama-ocr/lib/python3.10/site-packages/PIL/Image.py", line 3532, in open
    raise UnidentifiedImageError(msg)
```

**Comment by imanoop7** [comment]
will work on it, busy with some other projects

**Comment by nigelp** [comment]
Same + 2

**Comment by imanoop7** [comment]
pls test it and add your review,
pls create a new issue if you face any other problem in future

**Comment by imanoop7** [comment]
pls check now

**Comment by nigelp** [comment]
Almost. :) It now appears to see the file, but still no processing. Instead I get - 

Error processing image: 404 Client Error: Not Found for url: http://localhost:11434/api/generate

Tried with 2 LLMs

**Comment by imanoop7** [comment]
Pls check if the ollama is up and running and you have downloaded the model

**Comment by nigelp** [comment]
Yes I have a LLava installed and up and running, and the model exists. I know this because I just tested it with a JPEG and it worked perfectly. It's just PDFs that give a problem. What's happening now is the llava 7B model  is going through a process but at the end of it it just hallucinates, it doesn't give me what's actually in the PDF. it's weird.

I've just done a series of different tests with different PDFs and settings (e.g. pre-processing off and also a test with a custom prompt added). At no point has the app even suggested it's seen the file contents at all. Tried with Llama 11b vision too. Two example fail messages for you. :)

![Image](https://github.com/user-attachments/assets/a0807cda-1855-4af5-84b9-7261db780ef4)
![Image](https://github.com/user-attachments/assets/b86aaf27-2316-4fc2-b08a-1e191a13010d)

**Comment by nigelp** [comment]
Just noticed that your default prompt says "look at this image" could that be the problem?

**Comment by imanoop7** [comment]
> Yes I have a LLava installed and up and running, and the model exists. I know this because I just tested it with a JPEG and it worked perfectly. It's just PDFs that give a problem. What's happening now is the llava 7B model is going through a process but at the end of it it just hallucinates, it doesn't give me what's actually in the PDF. it's weird.
> 
> I've just done a series of different tests with different PDFs and settings (e.g. pre-processing off and also a test with a custom prompt added). At no point has the app even suggested it's seen the file contents at all. Tried with Llama 11b vision too. Two example fail messages for you. :)
> 
> ![Image](https://github.com/user-attachments/assets/a0807cda-1855-4af5-84b9-7261db780ef4) ![Image](https://github.com/user-attachments/assets/b86aaf27-2316-4fc2-b08a-1e191a13010d)

-------------------
sometimes llava does not give proper answer for certain output type, try llama3.2-vision or granite one, these models are good for text extraction.

--
the error might coming because system does not have enough memory to run the model.

**Comment by nigelp** [comment]
Yes Granite worked. Thanks. All good now. :)
